### PR TITLE
add onlyprivateuse1 decorator for test framework

### PR DESCRIPTION
**Pull request by Bin1024**
Fixes #ISSUE_NUMBER
The current community testing framework does not have a decorator for privateuse1, we have fixed this

### CODE REVIEW

File: torch/testing/_internal/common_device_type.py [hunk 1]
```diff
@@ -1231,6 +1239,12 @@ def onlyCUDA(fn):
 def onlyMPS(fn):
     return onlyOn('mps')(fn)
 
+def onlyPRIVATEUSE1(fn):
+    device_type = torch._C._get_privateuse1_backend_name()
+    device_mod = getattr(torch, device_type, None)
+    assert device_mod is not None, f'''torch has no module of `{self.device_type}`, you should register
```
Comment (albanD) [inline torch/testing/_internal/common_device_type.py:1245]: I feel like this function should work even when there is no private namespace available: it should just skip the test.

**Review by albanD — COMMENTED** [review]
Sounds good.
We should make sure that these functions always work so that they can be used in generic code.

File: torch/testing/_internal/common_device_type.py (hunk 1 above)
Comment (Bin1024) [reply torch/testing/_internal/common_device_type.py]: Yeah，I‘ve fixed it.

**Comment by Bin1024** [comment]
Hi @albanD ， I've fixed it. Please take a moment to review it again

**Comment by Bin1024** [comment]
@albanD hi sorry to bother you,could you take a look?

File: torch/testing/_internal/common_device_type.py [hunk 2]
```diff
@@ -1065,6 +1070,10 @@ def only_fn(slf, *args, **kwargs):
             if self.device_type != slf.device_type:
                 reason = "Only runs on {0}".format(self.device_type)
                 raise unittest.SkipTest(reason)
+            device_mod = getattr(torch, self.device_type, None)
+            if device_mod is None:
```
[the remark this answers is not in the document]
Comment (Bin1024) [reply torch/testing/_internal/common_device_type.py:1074]: I have tested it. It doesn't skip cpu tests.

[the remark this answers is not in the document]
Comment (albanD) [reply torch/testing/_internal/common_device_type.py:1074]: Ho we did add `torch.cpu`. But we don't have `torch.xla`.
Would it be possible to restrict this skip to privateuseone setup only? I don't want to silently skip things on other devices,

[the remark this answers is not in the document]
Comment (Bin1024) [reply torch/testing/_internal/common_device_type.py]: I have fixed it. Is this feasible?

**Review by albanD — APPROVED** [review]
Thanks for the update. Sounds ok even though I wish we could have test for it.

**Comment by Bin1024** [comment]
Hi @albanD , I don't understand why CI has failed. Can you help me?

**Comment by albanD** [comment]
@pytorchbot merge -r

These look unrelated. Hopefully the rebase will fix it

**Comment by albanD** [comment]
@pytorchbot merge -i

These look very much unrelated.

**Pull request closed without merging**